### PR TITLE
[TTAHUB-1837] Add collaborator permission check

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Collaborators need the ability to view all events in their region. In addition they need to view and edit any reports they collaborate on. This change modifies the service call to get the appropriate events for the user.

## How to test

- If a user has read/write training reports they should see events for all regions and be able to edit them.
- If a user is a collaborator they should see all reports in their region and be able to edit any report they collaborate on.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1837

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: src/routes/events/handlers.js [hunk 1]
```diff
@@ -79,7 +98,7 @@ export const getHandler = async (req, res) => {
 
     const auth = await getEventAuthorization(req, res, event);
 
-    if (!auth.canRead()) {
+    if (!auth.canRead() && !auth.isCollaborator()) {
```
Comment (GarrettEHill) [inline src/routes/events/handlers.js:101]: Should the isCollaborator be specific to the TR reports or will this also indicate if they are a collaborator on the AR?

File: src/services/accessValidation.js [hunk 2]
```diff
@@ -63,6 +88,26 @@ export async function getUserReadRegions(userId) {
...
 }
 
+export async function getUserTrainingReportReadRegions(userId) {
+  try {
+    const readRegions = await Permission.findAll({
+      attributes: ['regionId'],
+      where: {
+        userId,
+        [Op.or]: [
+          { scopeId: SCOPES.READ_WRITE_TRAINING_REPORTS },
```
Comment (GarrettEHill) [inline src/services/accessValidation.js:98]: Why Or all of these instead of the in operation?

File: src/routes/events/handlers.js (hunk 1 above)
Comment (thewatermethod) [reply src/routes/events/handlers.js:101]: this auth refers only to the TR, its a ref to the EventPolicy

File: src/services/accessValidation.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply src/services/accessValidation.js]: Looks like Matt has cleaned this up.

**Review by thewatermethod — APPROVED** [review]
Confirmed everything functions in the UI

**Pull request merged**